app/src/main/java/org/kontalk/ui/view/ComposerBar.java

```diff
@@ -201,6 +201,12 @@ public void afterTextChanged(Editable s) {
 
                 if (mListener != null)
                     mListener.textChanged(s);
+
+                // covert ascii to emojis if preference set
+                if (Preferences.getEmojiConverter(mContext)) {
+                    s = smileyConvert(s);
```

Comment:
This doesn't work. You're replacing the original "s" variable here, which has no effect. You should edit the "s" variable directly instead. Please note that any change to "s" will trigger the listener **again**, so you need to disable the listener while you do it.
See also: https://developer.android.com/reference/android/text/TextWatcher.html#afterTextChanged(android.text.Editable)